script/tool/lib/src/test_command.dart

```diff
@@ -32,72 +32,64 @@ class TestCommand extends PluginCommand {
       'This command requires "flutter" to be in your path.';
 
   @override
-  Future<void> run() async {
-    final List<String> failingPackages = <String>[];
-    await for (final Directory packageDir in getPackages()) {
-      final String packageName =
-          p.relative(packageDir.path, from: packagesDir.path);
-      if (!packageDir.childDirectory('test').existsSync()) {
-        print('SKIPPING $packageName - no test subdirectory');
-        continue;
-      }
+  Future<PackageResult> runForPackage(Directory package) async {
+    if (!package.childDirectory('test').existsSync()) {
+      return PackageResult.skip('No test/ directory.');
+    }
 
-      print('RUNNING $packageName tests...');
+    bool passed;
+    if (isFlutterPackage(package)) {
+      passed = await _runFlutterTests(package);
+    } else {
+      passed = await _runDartTests(package);
+    }
+    return passed ? PackageResult.success() : PackageResult.fail();
+  }
 
-      final String enableExperiment = getStringArg(kEnableExperiment);
+  /// Runs the Dart tests for a Flutter package, returning true on success.
+  Future<bool> _runFlutterTests(Directory package) async {
+    final String experiment = getStringArg(kEnableExperiment);
 
-      // `flutter test` automatically gets packages.  `pub run test` does not.  :(
-      int exitCode = 0;
-      if (isFlutterPackage(packageDir)) {
-        final List<String> args = <String>[
-          'test',
-          '--color',
-          if (enableExperiment.isNotEmpty)
-            '--enable-experiment=$enableExperiment',
-        ];
+    final int exitCode = await processRunner.runAndStream(
+      flutterCommand,
+      <String>[
+        'test',
+        '--color',
+        if (experiment.isNotEmpty) '--enable-experiment=$experiment',
+        if (isWebPlugin(package)) '--platform=chrome',
```

Comment:
Can we add a deprecation warning to this, so people stop doing flutter test --platform=chrome? Ideally, this should be deleted once the last couple of plugins are migrated to flutter drive (soon)

Reply:
Added a comment.